docs(CSS): Clarify style() container query support, simplify examples (#24481)
mdn/content: files/en-us/web/css/css_container_queries/index.md

{{CSSRef}}

> Container queries allow us to look at an element and apply styles to the children of that element based on the size of their container.
> This is an alternative to [media queries](/en-US/docs/Web/CSS/Media_Queries) which apply styles to elements based on viewport size or other device characteristics.
> If a container has less available space in the surrounding context, you can hide certain elements or use smaller fonts, for example.

Reviewer: ```suggestion
Container queries enable you to apply styles to an element based on the size of the element's container. If, for example, a container has less space available in the surrounding context, you can hide certain elements or use smaller fonts.
Container queries are an alternative to [media queries](/en-US/docs/Web/CSS/Media_Queries), which apply styles to elements based on viewport size or other device characteristics.

```

{{CSSRef}}

Container queries enable you to apply styles to an element based on the size of the element's container. If, for example, a container has less space available in the surrounding context, you can hide certain elements or use smaller fonts.
Container queries are an alternative to [media queries](/en-US/docs/Web/CSS/Media_Queries), which apply styles to elements based on viewport size or other device characteristics.

> ![A webpage with three card components displayed in different layouts depending on their container's size.](container-query.svg)

Author: ```suggestion
![A media query based on the viewport's width, which is the full width of the browser, and a container query based on the width of a container context, which is the width of the container element.](container-query.svg)
```

  Applies layout, style, and inline-size containment to the element.
- `normal`: The element is not a query container for any container size queries, but remains a query container for container style queries.

> Take the following example which is a card component for a blog post with a title and some text:

Reviewer: ```suggestion
Consider the following example of a card component for a blog post with a title and some text:
```

```html
<div class="post">
  <div class="card">
    <h2>Card title</h2>
    <p>Card content</p>
  </div>
</div>
```

You can create a containment context using the `container-type` property:

```css
.post {
  container-type: inline-size;
}
```

> An {{cssxref("@container")}} at-rule allows you to write a container query.

Reviewer: ```suggestion
Next, use the {{cssxref("@container")}} at-rule to define a container query.
```

  font-size: 1em;
}

/* if the container is larger than 700px */
@container (min-width: 700px) {
  .card h2 {
    font-size: 2em;
  }
}
```

Using container queries, the card can be reused in multiple areas of a page without needing to know specifically where it will be placed each time.

> If there is a smaller container, the card title will be smaller, and if there is a larger container with more space available, the card title will be larger.

Reviewer: ```suggestion
If the container with the card is narrower than `700px`, the font of the card title will be small, and if the card is in a container that's wider than `700px`, the font of the card title will be bigger.
```
